### PR TITLE
Add directory buttons

### LINKED ISSUE #2944: Open directory buttons
- Themes (sadly the theme selection is already pretty small, and under windows there is also another option "open console"),
so there is just no place anymore, any ideas @def- ?

- Skins

- Demos

### PR DESCRIPTION
**Pull request by Jupeyy**
fixes #2944

windows:

screenshot_2020-09-29_03-16-32

skin:

screenshot_2020-09-29_03-20-32

but no chance for demo:
screenshot_2020-09-29_03-23-13

naming suggestions are welcome
theme directory, themes directory
whatever

### CODE REVIEW

**Comment by Ekidoxx** [comment]
Definitely _Theme*s* directory_ & _Skin*s* Database_

What about making the buttons having same spaces around the text, rather than all of them being the same size? Like how it's almost always done on any website...

Anyway, if the scrollbar area would be shortened a bit, it would allow having 2nd line of buttons where fetch data button could be moved together with newly created _Demos directory_, and it should also help with translations that have longer texts

**Comment by Jupeyy** [comment]
the skin database button was added by deen and opens the actually database on the website, this is for easier acces to customize skins, bcs most ppl in the community still use the data directory to replace stuff.

Would ofc be nice to have a perfect fitting text, but its hard with translations, would require quite a bit of reprogramming(bcs its not easy for a constant margin), even tho u are right, that this would look much better :D 

Yeah, i'll wait for deen about the 2nd line, but i guess that will be required then

**Comment by fokkonaut** [comment]
This should open config directory paths, does it?

**Comment by Jupeyy** [comment]
if you have a file explorer installed, yes

**Comment by Ekidoxx** [comment]
(I meant missing `s` in buttons for Skins Database & Themes. Sorry, could have been more clear)

**Comment by def-** [comment]
"Open theme directory" is not working, and is connected with "Dynamic Camera" button.

**Comment by Jupeyy** [comment]
strange, cannot reproduce

File: src/game/client/components/menus_settings.cpp [hunk 1]
```diff
@@ -245,13 +243,28 @@ void CMenus::RenderSettingsGeneral(CUIRect MainView)
...
+		Left.HSplitTop(15.0f, 0, &Left);
+		CUIRect DirectoryButton;
+		Left.HSplitBottom(25.0f, &Left, &DirectoryButton);
 		RenderThemeSelection(Left);
 
+		DirectoryButton.HSplitTop(5.0f, 0, &DirectoryButton);
+		if(DoButton_Menu(&DirectoryButton, Localize("Open theme directory"), 0, &DirectoryButton))
+		{
+			char aBuf[MAX_PATH_LENGTH];
+			char aBufFull[MAX_PATH_LENGTH];
```
Comment (def-) [inline src/game/client/components/menus_settings.cpp:255]: I think this should be MAX_PATH_LENGTH+7

Comment (Jupeyy) [reply src/game/client/components/menus_settings.cpp:255]: true

**Comment by def-** [comment]
> strange, cannot reproduce

Fixed in https://github.com/ddnet/ddnet/pull/2947

**Comment by Jupeyy** [comment]
So, "Skins database" @def- ?
sounds a bit weird xD, almost as the skines are owning the database

**Comment by def-** [comment]
It's called "Skin Database". "Skins directory" is fine.

**Pull request merged**